### PR TITLE
develop: update the third party support table

### PR DESCRIPTION
**Pull request by Mini256**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-third-party-support.md [hunk 1]
```diff
@@ -177,6 +184,7 @@ If you encounter problems when connecting to TiDB using the tools listed in this
 
 ## GUI
 
-| GUI | Latest tested version | Support level | Tutorial |
-| - | - | - | - |
-| [DBeaver](https://dbeaver.io/) | 23.0.3 | Full | [Connect to TiDB with DBeaver](/develop/dev-guide-gui-dbeaver.md) |
+| GUI                                                       | Latest tested version | Support level | Tutorial                                                                      |
+|-----------------------------------------------------------|-----------------------|---------------|-------------------------------------------------------------------------------|
+| [JetBrains DataGrip](https://www.jetbrains.com/datagrip/) | 2023.2.1              | Full          | [Connect to TiDB with JetBrains DataGrip](/develop/dev-guide-gui-datagrip.md) |
+| [DBeaver](https://dbeaver.io/)                            | 23.0.3                | Full          | [Connect to TiDB with DBeaver](/develop/dev-guide-gui-dbeaver.md)             |
```
Comment (Icemap) [inline develop/dev-guide-third-party-support.md:190]:
```suggestion
| [DBeaver](https://dbeaver.io/)                            | 23.0.3                | Full          | [Connect to TiDB with DBeaver](/develop/dev-guide-gui-dbeaver.md)             |
| [VS Code](https://code.visualstudio.com/)                            | 1.72.0                | Full          | [Connect to TiDB with Visual Studio Code](/develop/dev-guide-gui-vscode-sqltools.md)             |
```

**Review by Icemap — APPROVED** [review]
Rest LGTM

File: develop/dev-guide-third-party-support.md [hunk 2]
```diff
@@ -177,6 +184,8 @@ If you encounter problems when connecting to TiDB using the tools listed in this
...
+| [VS Code](https://code.visualstudio.com/)                            | 1.72.0                | Full          | [Connect to TiDB with Visual Studio Code](/develop/dev-guide-gui-vscode-sqltools.md)             |
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-support.md:191]:
```suggestion
| [Visual Studio Code](https://code.visualstudio.com/)                            | 1.72.0                | Full          | [Connect to TiDB with Visual Studio Code](/develop/dev-guide-gui-vscode-sqltools.md)             |
```

**Review by Oreoxmt — APPROVED** [review]

**Comment by enkilee** [comment]
\assign

**Pull request merged**